frontend/src/components/TransactionItem.tsx

```diff
@@ -191,52 +191,143 @@ function TransactionItem({ tx }: Props) {
               </div>
             )}
           </CredenzaBody>
-          <CredenzaFooter className="!justify-start mt-4 !flex-col">
-            <div
-              className="flex items-center gap-2 cursor-pointer"
-              onClick={() => setShowDetails(!showDetails)}
-            >
-              Details
-              {showDetails ? (
-                <ChevronUp className="w-4 h-4" />
-              ) : (
-                <ChevronDown className="w-4 h-4" />
-              )}
-            </div>
-            {showDetails && (
-              <>
-                <div className="mt-6 !ml-0">
-                  <p>Preimage</p>
-                  <div className="flex items-center gap-4">
-                    <p className="text-muted-foreground break-all">
-                      {tx.preimage}
-                    </p>
-                    <CopyIcon
-                      className="cursor-pointer text-muted-foreground w-6 h-6"
-                      onClick={() => {
-                        if (tx.preimage) {
-                          copy(tx.preimage);
-                        }
-                      }}
-                    />
+          <CredenzaFooter>
+            <div className="mt-4 w-full">
+              <div
+                className="flex items-center gap-2 cursor-pointer"
+                onClick={() => setShowDetails(!showDetails)}
+              >
+                Details
+                {showDetails ? (
+                  <ChevronUp className="w-4 h-4" />
+                ) : (
+                  <ChevronDown className="w-4 h-4" />
+                )}
+              </div>
+              {showDetails && (
+                <>
+                  {tx.metadata?.tlv_records.map((record) => {
+                    if (record.type === 34349334) {
+                      // Chat message
+                      return (
+                        <div className="mt-6" key={record.type}>
+                          <p>Message</p>
+                          <p className="text-muted-foreground break-all">
+                            {hexToString(record.value)}
+                          </p>
+                        </div>
+                      );
+                    } else if (record.type === 7629169) {
```

Comment:
can this be a separate component?
to make this a bit smaller here (also if we have more)

Reply:
> to make this a bit smaller here

Yup I wanted to but wasn't sure where exactly to break it into a different component

Also do you mean a component just for the podcasting info?

Reply:
yep, I thought for the type 7629169

Reply:
Done!